repo/repo.go

```diff
@@ -39,29 +37,51 @@ type Options struct {
 
 	BlobMetricFunc BlobMetricFunc
 	QuotaManager   *quota.Manager
-	FsyncWarning   *sync.Once
+  FsyncWarning   *sync.Once
+
+  // If set, we will panic when an internal server error happens. This
+	// makes it easier to debug such errors.
+	GroupReadable bool
+
+	dirMode  os.FileMode
+	fileMode os.FileMode
 }
 
-// DefaultDirMode is the file mode used for directory creation if not
+// DefaultGroupReadable shows if files and dirs written should be group readable
 // overridden in the Options
+const DefaultGroupReadable = false
+
+// DefaultDirMode is the file mode used for directory creation by default
 const DefaultDirMode os.FileMode = 0700
 
-// DefaultFileMode is the file mode used for file creation if not
-// overridden in the Options
+// DefaultFileMode is the file mode used for file creation by default
 const DefaultFileMode os.FileMode = 0600
 
+// GroupReadableDirMode is the file mode used for directory creation if group readable
+const GroupReadableDirMode os.FileMode = 0750
```

Comment:
I'd prefer `0770` and `0660` an leave further restrictions to the umask used to run `rest-server`.